inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp

```diff
@@ -137,6 +147,63 @@ static void Transformation(ICNNNetwork::Ptr& clonedNetwork, const Config& conf)
+                    return rnn_cell->get_clip() != 0.0f;
+                }
+            return false;
+            });
+
+    pass_config->set_callback<ngraph::pass::GRUCellDecomposition>(
+            [](const_node_ptr &node) -> bool {
+                if (const auto &gru_cell = std::dynamic_pointer_cast<const ngraph::opset4::GRUCell>(node)) {
+                    return gru_cell->get_clip() != 0.0f || gru_cell->get_activations()
+                                                           != std::vector<std::string>{"sigmoid", "tanh"};
+                }
+                return false;
+            });
+
+    pass_config->set_callback<ngraph::pass::LSTMCellDecomposition>(
+            [](const_node_ptr &node) -> bool {
+                if (const auto &lstm_cell = std::dynamic_pointer_cast<const ngraph::opset4::LSTMCell>(node)) {
+                    return lstm_cell->get_clip() != 0.0f || lstm_cell->get_activations()
+                                                            != std::vector<std::string>{"sigmoid", "tanh", "tanh"};
+                }
+                return false;
+            });
+
+    pass_config->set_callback<ngraph::pass::ConvertTensorIteratorToRNNSequence,
+                              ngraph::pass::ConvertTensorIteratorToLSTMSequence,
+                              ngraph::pass::ConvertTensorIteratorToGRUSequence>(
+            [](const_node_ptr &node) -> bool {
+                if (auto ti_op = std::dynamic_pointer_cast<const ngraph::op::TensorIterator>(node)) {
+                    size_t count_rnn = 0;
+                    bool enable_convert_to_sequence = true;
+                    for (const auto &op : ti_op->get_body()->get_ops()) {
+                        if (const auto &rnn_cell = std::dynamic_pointer_cast<const ngraph::opset4::RNNCell>(op)) {
+                            enable_convert_to_sequence &= rnn_cell->get_clip() == 0.0f;
+                            count_rnn++;
+                        } else if (const auto &gru_cell = std::dynamic_pointer_cast<const ngraph::opset4::GRUCell>(
+                                op)) {
+                            enable_convert_to_sequence &= gru_cell->get_clip() == 0.0f;
+                            enable_convert_to_sequence &=
+                                    gru_cell->get_activations() == std::vector<std::string>{"sigmoid", "tanh"};
+                            count_rnn++;
+                        } else if (const auto &lstm_cell = std::dynamic_pointer_cast<const ngraph::opset4::LSTMCell>(
+                                op)) {
+                            enable_convert_to_sequence &= lstm_cell->get_clip() == 0.0f;
+                            enable_convert_to_sequence &=
+                                    lstm_cell->get_activations() == std::vector<std::string>{"sigmoid", "tanh", "tanh"};
+                            count_rnn++;
+                        }
+                    }
```

Comment:
These conditions are the same as for Cell decomposition passes. So we can extract it into separate SupportedCell checker.

Reply:
done